azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -9,7 +9,7 @@
 import six
 import abc
 import logging
-from azure.iot.device.iothub.transport import MQTTTransport
+from azure.iot.device.iothub.pipeline.pipeline_adapter import PipelineAdapter
```

Comment:
Probably should import from the package instead of the module, so the internal implementation of the package can be more easily changed.

Module specific imports within a package are only useful IMO if the package is a loose collection of unrelated modules (e.g. tooling package) or there is some module/subpackage within the package intended to exist separately from the main interface (e.g. importing from package.models)